### PR TITLE
Add a recipe for apache-airflow-providers-common-io

### PR DESCRIPTION
**Pull request by xylar**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge/help-python, this is ready for review.

The Windows error is expected, since apache-airflow does not support Windows.

**Comment by xylar** [comment]
Needed for:
https://github.com/conda-forge/airflow-feedstock/pull/126

File: recipes/apache-airflow-providers-common-io/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  sha256: 9a8e3a9538e5281bbccd2d3a86b1bbb180734e17a959090cde5e7758dded97a3
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
+    - python >=3.8,<4.dev0
```
Comment (ocefpaf) [inline recipes/apache-airflow-providers-common-io/meta.yaml:19]: Should it be the same as host?

Comment (xylar) [reply recipes/apache-airflow-providers-common-io/meta.yaml:19]: Oops, nice catch.

**Comment by xylar** [comment]
@ocefpaf, would you take another look when you have a chance?  Thank you!

**Comment by xylar** [comment]
@ocefpaf, much appreciated!

**Pull request merged**